asset-transfer-secured-agreement/application-javascript/app.js

```diff
@@ -0,0 +1,563 @@
+		}
+	}
+}
+
+async function readAsset(assetKey, ownerOrg, contractOrg1, contractOrg2) {
+	console.log(`${GREEN}--> Evaluate Transactions: ReadAsset, - ${assetKey} should be owned by ${ownerOrg}${RESET}`);
+	let resultBuffer;
+	resultBuffer = await evaluateTransaction(contractOrg1, assetKey);
+	checkAsset('Org1', resultBuffer, ownerOrg);
+	resultBuffer = await evaluateTransaction(contractOrg2, assetKey);
+	checkAsset('Org2', resultBuffer, ownerOrg);
+}
+
+// This application uses fabric-samples/test-network based setup and the companion chaincode
+// For this illustration, both Org1 & Org2 client identities will be used, however
+// notice they are used by two different "gateway"s to simulate two different running
+// applications from two different organizations.
+async function main() {
+	console.log(`${GREEN} **** START ****${RESET}`);
+	try {
+		const randomNumber = Math.floor(Math.random() * 100) + 1;
+		// use a random key so that we can run multiple times
+		const assetKey = `asset-${randomNumber}`;
+
+		/** ******* Fabric client init: Using Org1 identity to Org1 Peer ******* */
+		const gatewayOrg1 = await initGatewayForOrg1();
+		const networkOrg1 = await gatewayOrg1.getNetwork(channelName);
+		const contractOrg1 = networkOrg1.getContract(chaincodeName);
+
+		/** ******* Fabric client init: Using Org2 identity to Org2 Peer ******* */
+		const gatewayOrg2 = await initGatewayForOrg2();
+		const networkOrg2 = await gatewayOrg2.getNetwork(channelName);
+		const contractOrg2 = networkOrg2.getContract(chaincodeName);
+
+		try {
+			let transaction;
+
+			try {
+				// Create an asset by organization Org1, this only requires the owning
+				// organization to endorse.
+				// With the gateway using discovery, we should limit the organizations used
+				// to endorse. This only requires knowledge of the Organizations and not
+				// the actual peers that may be active at any given time.
+				const asset_properties = {
+					object_type: 'asset_properties',
+					asset_id: assetKey,
+					color: 'blue',
+					size: 35,
+					salt: Buffer.from(randomNumber.toString()).toString('hex')
+				};
+				const asset_properties_string = JSON.stringify(asset_properties);
+				console.log(`${GREEN}--> Submit Transaction: CreateAsset, ${assetKey} as Org1 - endorsed by Org1${RESET}`);
+				console.log(`${asset_properties_string}`);
+				transaction = contractOrg1.createTransaction('CreateAsset');
+				transaction.setEndorsingOrganizations(org1);
+				transaction.setTransient({
+					asset_properties: Buffer.from(asset_properties_string)
+				});
+				await transaction.submit(assetKey, `Asset ${assetKey} owned by ${org1} is not for sale`);
+				console.log(`*** Result: committed, asset ${assetKey} is owned by Org1`);
+			} catch (createError) {
+				console.log(`${RED}*** Failed: CreateAsset - ${createError}${RESET}`);
+			}
+
+			// read the public details by both orgs
+			await readAsset(assetKey, org1, contractOrg1, contractOrg2);
```

Comment:
I see what you are doing here, reading it twice for test purpose, but as a regular application user I'd expect readAsset() to only read it once. It is not expected that readAsset() would read it twice. Maybe just change the function name to readAssetByBothOrgs() to make it clear.

Reply:
done